Parser/pegen.c

```diff
@@ -410,7 +410,49 @@ get_error_line(Parser *p, Py_ssize_t lineno)
 }
 
 Py_ssize_t
-_PyPegen_byte_offset_to_character_offset(PyObject *line, Py_ssize_t col_offset)
+byte_offset_difference_for_unicode(Py_ssize_t size, PyObject *text)
+{
+    PyObject *unicode_data = PyImport_ImportModuleNoBlock("unicodedata");
+    if (!unicode_data) {
+        return -1;
+    }
+
+    PyObject *east_asian_width = PyObject_GetAttrString(unicode_data, "east_asian_width");
+    Py_DECREF(unicode_data);
+    if (!east_asian_width) {
+        return -1;
+    }
+
+    PyObject *current, *width = NULL;
+    Py_ssize_t i, difference = 0;
+    for (i = 0; i < size; i++) {
+        current = PySequence_GetItem(text, i);
+        if (!current) {
+            goto error;
+        }
+
+        PyObject *width = PyObject_CallOneArg(east_asian_width, current);
+        Py_DECREF(current);
+        if (!width) {
+            goto error;
+        }
+
+        if (_PyUnicode_EqualToASCIIString(width, "W")) {
```

Comment:
We should probably use Martin v. Löwis's [provided algorithm](https://bugs.python.org/file24773/width.py) from this [issue](https://bugs.python.org/issue12568).

(It just adds `"F"` to the width specifiers that should be doubled.)

Reply:
Oh, missed that one. Thanks, will add `F` as well.